Markdown fails to load with "expecting a single list" because the markdown file is actually invalid

**Describe the bug**
Some specific invalid markdown needs to be handled inside challenges. For example, in [iterating-over-data/problem-set-while-loops.md](https://github.com/Ada-Developers-Academy/core-unit-1/blob/4efb1c7a9f85d6590ab83cce6102f9aee963e266/iterating-over-data/problem-set-while-loops.md?plain=1#L265) the GitHub-flavored markdown parser makes a mess of challenge `0a8cfa41-b770-4e80-8af0-f4d5127c8424`. (You can see the mess when you view the markdown directly on GitHub, too.) This is because there aren't spaces indenting the rest of the code block after the list item marker.

That is to say, the markdown contains this invalid syntax:

~~~md
* ```python
for counter in range(4):
    print(counter)
```
~~~

whereas valid syntax would indent the rest of the code block to contain it in the list item:

~~~md
* ```python
  for counter in range(4):
      print(counter)
  ```
~~~

Learn must be handling the invalid markdown just fine. Unfortunatley.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to https://learn-viewer.cyvaer.com/?course=https%3A%2F%2Fraw.githubusercontent.com%2FAda-Developers-Academy%2Fcore%2Fmain%2Fcourse.yaml&section=Unit+1&standard=giOpFA&content-file-uid=3302a20d
1. See "Markdown failed to load"

On the console, this error appears:
```
index-248cf47d.js:40 Error: Expected a single list
```

**Expected behavior**
The challenge should display with the code blocks inside of options.

--- a/package.json
+++ b/package.json
@@ -25,7 +25,8 @@
     "remark-gfm": "^3.0.1",
     "unified": "^10.1.2",
     "unist-builder": "^4.0.0",
-    "unist-util-find-after": "^5.0.0"
+    "unist-util-find-after": "^5.0.0",
+    "unist-util-is": "^6.0.0"
   },
   "devDependencies": {
     "@types/js-yaml": "^4.0.5",

--- a/package.json
+++ b/package.json
@@ -36,6 +36,7 @@
     "@typescript-eslint/eslint-plugin": "^6.0.0",
     "@typescript-eslint/parser": "^6.0.0",
     "@vitejs/plugin-react-swc": "^3.3.2",
+    "dedent": "^1.5.1",
     "eslint": "^8.45.0",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.3",

--- a/src/components/document.tsx
+++ b/src/components/document.tsx
@@ -8,6 +8,7 @@ import remarkCallout from '../remark-plugins/remark-callout-plugin';
 import remarkChallenges from '../remark-plugins/remark-challenge-plugin';
 import remarkFixUrls from '../remark-plugins/remark-fix-urls-plugin';
 import remarkFixListsWithCode from '../remark-plugins/remark-fix-lists-with-code-plugin';
+import { fixBadCodeIndentInList } from '../remark-plugins/preprocess-fix-bad-code-indent';
 
 import '../remark-plugins/remark-callout-plugin.css';
 import { ClearMarkdownAction } from '../reducers/markdown-actions';
@@ -54,6 +55,10 @@ function Document() {
     );
   }
 
+  // Preprocess the markdown.
+  // Some markdown files have bad code indents that need to be fixed up before parsing.
+  const renderableMarkdown = fixBadCodeIndentInList(mdState.currentMarkdown);
+
   return (
     <div id="markdown-container">
       <div className='right-align'>
@@ -90,7 +95,7 @@ function Document() {
             }
           }}
         >
-          {mdState.currentMarkdown}
+          { renderableMarkdown }
         </ReactMarkdown>
       </ErrorBoundary>
     </div>

--- a/src/components/document.tsx
+++ b/src/components/document.tsx
@@ -7,6 +7,7 @@ import rehypeExternalLinks from 'rehype-external-links';
 import remarkCallout from '../remark-plugins/remark-callout-plugin';
 import remarkChallenges from '../remark-plugins/remark-challenge-plugin';
 import remarkFixUrls from '../remark-plugins/remark-fix-urls-plugin';
+import remarkFixListsWithCode from '../remark-plugins/remark-fix-lists-with-code-plugin';
 
 import '../remark-plugins/remark-callout-plugin.css';
 import { ClearMarkdownAction } from '../reducers/markdown-actions';
@@ -61,10 +62,11 @@ function Document() {
       <ErrorBoundary fallback={<div>The Markdown failed to load</div>}>
         <ReactMarkdown
           remarkPlugins={[
+            [remarkFixUrls, { rootUrl: mdState.markdownRootUrl }],
             remarkGfm,
+            remarkFixListsWithCode,
             remarkCallout,
             remarkChallenges,
-            [remarkFixUrls, { rootUrl: mdState.markdownRootUrl }]
           ]}
           rehypePlugins={[
             rehypeRaw,

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
@@ -1,0 +1,103 @@
+import { describe, it, expect } from 'vitest';
+import { fixBadCodeIndentInList } from "./preprocess-fix-bad-code-indent";
+
+const input: string = `
+* \`\`\`python
+for counter in range(4):
+    print(counter)
+\`\`\`
+
+* Another list item
+  with some content in it
+
+*
+Some stuff below an empty list
+
+* \`\`\`python
+for counter in range(5):
+    print(counter)
+\`\`\`
+
+* \`\`\`python
+for counter in range(5, 0):
+    print(counter)
+\`\`\`
+
+\`\`\`python
+A code block just hanging out
+\`\`\`
+
+* \`\`\`python
+for counter in range(5, 0, -1):
+    print(counter)
+\`\`\`
+`;
+
+const expected: string = `
+* \`\`\`python
+  for counter in range(4):
+      print(counter)
+  \`\`\`
+
+* Another list item
+  with some content in it
+
+*
+Some stuff below an empty list
+
+* \`\`\`python
+  for counter in range(5):
+      print(counter)
+  \`\`\`
+
+* \`\`\`python
+  for counter in range(5, 0):
+      print(counter)
+  \`\`\`
+
+\`\`\`python
+A code block just hanging out
+\`\`\`
+
+* \`\`\`python
+  for counter in range(5, 0, -1):
+      print(counter)
+  \`\`\`
+`;
+
+describe('preprocess-fix-bad-code-indent', () => {
+  it('fixes bad code indent in one list items', () => {
+    const input = `
+* \`\`\`python
+for counter in range(4):
+    print(counter)
+\`\`\`
+`;
+
+    const expected = `
+* \`\`\`python
+  for counter in range(4):
+      print(counter)
+  \`\`\`
+`;
+    const result = fixBadCodeIndentInList(input);
+    expect(result).toEqual(expected);
+  });
+
+  it('fixes bad code indent in many list items', () => {
+    const result = fixBadCodeIndentInList(input);
+    expect(result).toEqual(expected);
+  });
+
+  it('ignores already good code indent', () => {
+    const input = `
+* \`\`\`python
+  for counter in range(4):
+      print(counter)
+  \`\`\`
+`;
+
+    const result = fixBadCodeIndentInList(input);
+    expect(result).toEqual(input);
+  });
+});

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
@@ -100,4 +100,29 @@ for counter in range(4):
     const result = fixBadCodeIndentInList(input);
     expect(result).toEqual(input);
   });
+
+  it(`indents if the code block finishes on the next line`, () => {
+    const input = `
+* \`\`\`python for counter in range(4): print(counter)
+\`\`\`
+`;
+    const expected = `
+* \`\`\`python for counter in range(4): print(counter)
+  \`\`\`
+`;
+    const result = fixBadCodeIndentInList(input);
+    expect(result).toEqual(expected);
+  });
+
+  it(`correctly handles code blocks on a single line`, () => {
+    const input = `
+* \`\`\`python for counter in range(4): print(counter)\`\`\`
+\`\`\`
+def perfectly_valid_code():
+    return "because it's a separate code block.";
+\`\`\`
+`;
+    const result = fixBadCodeIndentInList(input);
+    expect(result).toEqual(input);
+  });
 });

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.test.ts
@@ -1,4 +1,5 @@
 import { describe, it, expect } from 'vitest';
+import dedent from 'dedent';
 import { fixBadCodeIndentInList } from "./preprocess-fix-bad-code-indent";
 
 const input: string = `
@@ -67,19 +68,19 @@ A code block just hanging out
 
 describe('preprocess-fix-bad-code-indent', () => {
   it('fixes bad code indent in one list items', () => {
-    const input = `
-* \`\`\`python
-for counter in range(4):
-    print(counter)
-\`\`\`
-`;
+    const input = dedent`
+      * \`\`\`python
+      for counter in range(4):
+          print(counter)
+      \`\`\`
+      `;
 
-    const expected = `
-* \`\`\`python
-  for counter in range(4):
-      print(counter)
-  \`\`\`
-`;
+    const expected = dedent`
+      * \`\`\`python
+        for counter in range(4):
+            print(counter)
+        \`\`\`
+      `;
     const result = fixBadCodeIndentInList(input);
     expect(result).toEqual(expected);
   });
@@ -90,38 +91,38 @@ for counter in range(4):
   });
 
   it('ignores already good code indent', () => {
-    const input = `
-* \`\`\`python
-  for counter in range(4):
-      print(counter)
-  \`\`\`
-`;
+    const input = dedent`
+      * \`\`\`python
+        for counter in range(4):
+            print(counter)
+        \`\`\`
+      `;
 
     const result = fixBadCodeIndentInList(input);
     expect(result).toEqual(input);
   });
 
   it(`indents if the code block finishes on the next line`, () => {
-    const input = `
-* \`\`\`python for counter in range(4): print(counter)
-\`\`\`
-`;
-    const expected = `
-* \`\`\`python for counter in range(4): print(counter)
-  \`\`\`
-`;
+    const input = dedent`
+      * \`\`\`python for counter in range(4): print(counter)
+      \`\`\`
+      `;
+    const expected = dedent`
+      * \`\`\`python for counter in range(4): print(counter)
+        \`\`\`
+      `;
     const result = fixBadCodeIndentInList(input);
     expect(result).toEqual(expected);
   });
 
   it(`correctly handles code blocks on a single line`, () => {
-    const input = `
-* \`\`\`python for counter in range(4): print(counter)\`\`\`
-\`\`\`
-def perfectly_valid_code():
-    return "because it's a separate code block.";
-\`\`\`
-`;
+    const input = dedent`
+      * \`\`\`python for counter in range(4): print(counter)\`\`\`
+      \`\`\`
+      def perfectly_valid_code():
+          return "because it's a separate code block.";
+      \`\`\`
+      `;
     const result = fixBadCodeIndentInList(input);
     expect(result).toEqual(input);
   });

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.ts
@@ -1,0 +1,33 @@
+export function fixBadCodeIndentInList(markdown: string): string {
+  // Match a list item that starts with a code block, followed by any number of lines that are missing
+  // indentation at two spaces, followed by the end of the code block.
+  const lines = markdown.split('\n');
+  let currentLine = 0;
+  while (currentLine < lines.length) {
+    currentLine = fixFirstBadCodeIndentInList(lines, currentLine);
+  }
+
+  markdown = lines.join('\n');
+  return markdown;
+}
+
+function fixFirstBadCodeIndentInList(lines: string[], startingLine: number): number {
+  const idxCodeStart = lines.findIndex((line, idx) => idx > startingLine && line.match(/^[*-] ```/));
+  if (idxCodeStart === -1) {
+    return lines.length;
+  }
+
+  if (lines[idxCodeStart].match(/^[*-] ```.*```/)) {
+    // The code block is on a single line, so there's nothing to fix. Skip to the next line after it.
+    return idxCodeStart + 1;
+  }
+
+  const idxCodeEnd = lines.findIndex((line, idx) => idx > idxCodeStart && line.match(/```/));
+  const needsIndent = lines.slice(idxCodeStart + 1, idxCodeEnd).some(line => !line.startsWith('  '));
+  if (needsIndent) {
+    for (let i = idxCodeStart + 1; i <= idxCodeEnd; i++) {
+      lines[i] = '  ' + lines[i];
+    }
+  }
+  return idxCodeEnd + 1;
+}

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.ts
@@ -12,7 +12,7 @@ export function fixBadCodeIndentInList(markdown: string): string {
 }
 
 function fixFirstBadCodeIndentInList(lines: string[], startingLine: number): number {
-  const idxCodeStart = lines.findIndex((line, idx) => idx > startingLine && line.match(/^[*-] ```/));
+  const idxCodeStart = lines.findIndex((line, idx) => idx >= startingLine && line.match(/^[*-] ```/));
   if (idxCodeStart === -1) {
     return lines.length;
   }

--- a/src/remark-plugins/preprocess-fix-bad-code-indent.ts
+++ b/src/remark-plugins/preprocess-fix-bad-code-indent.ts
@@ -23,7 +23,7 @@ function fixFirstBadCodeIndentInList(lines: string[], startingLine: number): num
   }
 
   const idxCodeEnd = lines.findIndex((line, idx) => idx > idxCodeStart && line.match(/```/));
-  const needsIndent = lines.slice(idxCodeStart + 1, idxCodeEnd).some(line => !line.startsWith('  '));
+  const needsIndent = lines.slice(idxCodeStart + 1, idxCodeEnd + 1).some(line => !line.startsWith('  '));
   if (needsIndent) {
     for (let i = idxCodeStart + 1; i <= idxCodeEnd; i++) {
       lines[i] = '  ' + lines[i];

--- a/src/remark-plugins/remark-challenge-plugin/index.ts
+++ b/src/remark-plugins/remark-challenge-plugin/index.ts
@@ -1,9 +1,12 @@
 import { visit, SKIP, type Visitor, type VisitorResult } from 'unist-util-visit';
+import { is } from 'unist-util-is';
 import { toMarkdown } from 'mdast-util-to-markdown';
 import { toString as mdastToString } from 'mdast-util-to-string';
+
 import type { Transformer } from 'unified';
 import type { Node, Parent } from 'unist';
-import type { Root, Heading, Text, ListItem } from 'mdast';
+import type { Root, Heading, Text, ListItem, Paragraph } from 'mdast';
+
 import { extractInfoNode, type ChallengeInfo } from './md-to-js-parse';
 import { extractOptionsNodesAndData, getList, unList } from './make-option';
 import { makeMdToHastNode } from './generate-node';
@@ -36,6 +39,7 @@ const visitor: Visitor<Node> = (node: Node, index: number | null, parent: Parent
 
       const answer = extractAnswer(childrenBetween);
       const answerIds = getAnswerIds(answer, challengeInfo, mdToIdMap);
+      challengeInfo.answer = answerIds;
 
       console.log("Options:", optionIds, "Answer:", answerIds);
 
@@ -71,22 +75,28 @@ function getAnswerIds(answer: Node[], challengeInfo: ChallengeInfo, mdToIdMap: M
     return [];
   }
 
-  const answerItems: ListItem[] = ['multiple-choice', 'checkbox'].includes(challengeInfo.challengeType) ? getList(answer) : [];
-  let answerIds: string[] = [];
+  function getAnswerId(node: Paragraph | ListItem): string {
+    const markdown = toMarkdown(unList(node));
+    if (!mdToIdMap.has(markdown)) {
+      return markdown; // If there is no answer ID in the map, just return the answer itself.
+    }
+    return mdToIdMap.get(markdown)!;
+  }
+
+  // If this is a single paragraph node, that's the answer.
+  if (answer.length === 1 && is(answer[0], 'paragraph')) {
+    return [getAnswerId(answer[0] as Paragraph)];
+  }
+
   try {
-    answerIds = answerItems.map(item => {
-      const markdown = toMarkdown(unList(item));
-      if (!mdToIdMap.has(markdown)) {
-        throw new Error(`Could not find an option with markdown: ${markdown}`);
-      }
-      return mdToIdMap.get(markdown)!;
-    });
-    challengeInfo.answer = answerIds;
+    const answerItems: ListItem[] = getList(answer);
+    const answerIds = answerItems.map(getAnswerId);
+    return answerIds;
   } catch(err: any) {
     console.error(`Failed to parse the answer for ${challengeInfo.id}. Error: "${err.message}"`);
   }
 
-  return answerIds;
+  return [];
 }
 
 const tagPair = (tagName: string) => {

--- a/src/remark-plugins/remark-challenge-plugin/index.ts
+++ b/src/remark-plugins/remark-challenge-plugin/index.ts
@@ -34,29 +34,8 @@ const visitor: Visitor<Node> = (node: Node, index: number | null, parent: Parent
       const [optionsRoot, optionIds, mdToIdMap] = extractOptionsNodesAndData(options, challengeInfo);
       challengeInfo.options = optionIds;
 
-      // Extract answer data (for multiple choice)
-      // The issue with getting answers is that they need to be correlated with the options so we can
-      // extract the *option IDs* that correspond to the answers. We have to compare the markdown of
-      // the answers to the markdown of the options.
-      //
-      // NOTE:
-      // When it comes time to implement `ordering` types, ONLY answers will exist. ALSO, they will
-      // be in an ordered list, not an unordered list. So we can't share this parsing logic for every type.
       const answer = extractAnswer(childrenBetween);
-      const answerItems: ListItem[] = ['multiple-choice', 'checkbox'].includes(challengeInfo.challengeType) ? getList(answer) : [];
-      let answerIds: string[] = [];
-      try {
-        answerIds = answerItems.map(item => {
-          const markdown = toMarkdown(unList(item));
-          if (!mdToIdMap.has(markdown)) {
-            throw new Error(`Could not find an option with markdown: ${markdown}`);
-          }
-          return mdToIdMap.get(markdown)!;
-        });
-        challengeInfo.answer = answerIds;
-      } catch(err: any) {
-        console.error(`Failed to parse the answer for ${challengeInfo.id}. Error: "${err.message}"`);
-      }
+      const answerIds = getAnswerIds(answer, challengeInfo, mdToIdMap);
 
       console.log("Options:", optionIds, "Answer:", answerIds);
 
@@ -79,11 +58,42 @@ const visitor: Visitor<Node> = (node: Node, index: number | null, parent: Parent
   }
 };
 
+// Extract answer data (for multiple choice)
+// The issue with getting answers is that they need to be correlated with the options so we can
+// extract the *option IDs* that correspond to the answers. We have to compare the markdown of
+// the answers to the markdown of the options.
+//
+// NOTE:
+// When it comes time to implement `ordering` types, ONLY answers will exist. ALSO, they will
+// be in an ordered list, not an unordered list. So we can't share this parsing logic for every type.
+function getAnswerIds(answer: Node[], challengeInfo: ChallengeInfo, mdToIdMap: Map<string, string>) {
+  if (answer.length === 0) {
+    return [];
+  }
+
+  const answerItems: ListItem[] = ['multiple-choice', 'checkbox'].includes(challengeInfo.challengeType) ? getList(answer) : [];
+  let answerIds: string[] = [];
+  try {
+    answerIds = answerItems.map(item => {
+      const markdown = toMarkdown(unList(item));
+      if (!mdToIdMap.has(markdown)) {
+        throw new Error(`Could not find an option with markdown: ${markdown}`);
+      }
+      return mdToIdMap.get(markdown)!;
+    });
+    challengeInfo.answer = answerIds;
+  } catch(err: any) {
+    console.error(`Failed to parse the answer for ${challengeInfo.id}. Error: "${err.message}"`);
+  }
+
+  return answerIds;
+}
+
 const tagPair = (tagName: string) => {
   const isTagStart = (node: Node) => isTag(node, tagName);
   const isTagEnd = (node: Node) => isEndTag(node, tagName);
   return [isTagStart, isTagEnd];
-}
+};
 
 const [isChallengeStart, isChallengeEnd] = tagPair('challenge');
 

--- a/src/remark-plugins/remark-challenge-plugin/index.ts
+++ b/src/remark-plugins/remark-challenge-plugin/index.ts
@@ -84,7 +84,7 @@ function getAnswerIds(answer: Node[], challengeInfo: ChallengeInfo, mdToIdMap: M
   }
 
   // If this is a single paragraph node, that's the answer.
-  if (answer.length === 1 && is(answer[0], 'paragraph')) {
+  if (answer.length === 1 && (is(answer[0], 'paragraph') || is(answer[0], 'code') || is(answer[0], 'text'))) {
     return [getAnswerId(answer[0] as Paragraph)];
   }
 

--- a/src/remark-plugins/remark-challenge-plugin/make-option.ts
+++ b/src/remark-plugins/remark-challenge-plugin/make-option.ts
@@ -1,5 +1,5 @@
 import { type Node } from 'unist';
-import { List, ListItem, Paragraph } from 'mdast';
+import { List, ListItem, Paragraph, Code, Text } from 'mdast';
 import { toMarkdown } from 'mdast-util-to-markdown';
 import { makeMdToHastNode, GeneratedNode } from './generate-node';
 import { ChallengeInfo } from './md-to-js-parse';
@@ -77,15 +77,22 @@ function makeCheckbox(option: Node, challenge: ChallengeInfo): GeneratedNode {
  * @param node A node that may be a Paragraph or a ListItem
  * @returns The Paragraph inside the ListItem or the Paragraph, depending on which was passed in.
  */
-export function unList(node: Node): Paragraph {
+export function unList(node: Node): Paragraph | Code | Text {
   if (node.type === 'paragraph') { return node as Paragraph; }
+  if (node.type === 'code') { return node as Code; }
+  if (node.type === 'text') { return node as Text; }
   if (node.type !== 'listItem') {
-    throw new Error('Expected a list or paragraph');
+    throw new Error('Expected a list or paragraph/code/text');
   }
 
   const listItem = node as ListItem;
-  if (listItem.children.length !== 1 && listItem.children[0].type !== 'paragraph') {
-    throw new Error('Expected a list with a single paragraph');
+  if (listItem.children.length !== 1) {
+    throw new Error('Expected a list with a single paragraph/code');
   }
-  return listItem.children[0] as Paragraph;
+
+  const firstchild = listItem.children[0];
+  if (firstchild.type === 'paragraph') { return firstchild as Paragraph; }
+  if (firstchild.type === 'code') { return firstchild as Code; }
+
+  throw new Error(`Expected a list with a single paragraph/code, but it had type '${firstchild.type}'`);
 }

--- a/src/remark-plugins/remark-fix-lists-with-code-plugin.ts
+++ b/src/remark-plugins/remark-fix-lists-with-code-plugin.ts
@@ -1,0 +1,58 @@
+/**
+ * A remark plugin that finds code blocks that are poorly formed and correctly nests them into the lists
+ * that they are supposed to be in.
+ */
+import { visit, CONTINUE, type Visitor, type VisitorResult } from 'unist-util-visit';
+import { is } from 'unist-util-is';
+import type { Transformer, Plugin } from 'unified';
+import type { Parent } from 'unist';
+import type { Root, Code, List } from 'mdast';
+
+interface FixCodeListOptions {
+}
+
+/**
+ * This plugin transforms relative URLs to absolute URLs. *
+ */
+export const plugin: Plugin<[FixCodeListOptions?], Root> = () => {
+  const visitor: Visitor<List> = (node: List, index: number | null, parent: Parent | null): VisitorResult => {
+    if (parent === null || index === null) return CONTINUE;
+
+    // The list has a list item, but it's empty. See if a code block comes after it as a sibling.
+    // If it does, move the code block into the list item.
+    if (is(node.children[0], 'listItem') && node.children[0].children.length === 0) {
+      const listItem = node.children[0];
+      if (is(parent.children[index + 1], 'code')) {
+        const code = parent.children[index + 1] as Code;
+        listItem.children.push(code);
+        parent.children.splice(index + 1, 1);
+      }
+    }
+  
+    return CONTINUE;
+  };
+
+  const listCoalescingVisitor: Visitor<List> = (node: List, index: number | null, parent: Parent | null): VisitorResult => {
+    if (parent === null || index === null) return CONTINUE;
+
+    // While this list is followed by other lists, coalesce them into this list.
+    while (is(parent.children[index + 1], 'list')) {
+      const nextList = parent.children[index + 1] as List;
+      node.children.push(...nextList.children);
+      parent.children.splice(index + 1, 1);
+    }
+
+    return CONTINUE;
+  };
+
+  const transformer: Transformer<Root, Root> = (tree: Root) => {
+    visit(tree, 'list', visitor);
+    visit(tree, 'list', listCoalescingVisitor);
+  };
+
+  return transformer;
+};
+
+
+
+export default plugin;

--- a/src/remark-plugins/remark-fix-urls-plugin.ts
+++ b/src/remark-plugins/remark-fix-urls-plugin.ts
@@ -20,7 +20,7 @@ export const plugin: Plugin<[FixUrlOptions?], Root> = (options) => {
 
     if (!rNode?.url.startsWith('http')) {
       const withoutFile = options?.rootUrl.slice(0, options?.rootUrl.lastIndexOf('/') + 1);
-      console.log(withoutFile, rNode.url);
+      console.log('Updating URL:', withoutFile, rNode.url);
       rNode.url = withoutFile + rNode.url;
     }
   

